src/ddmd/mtype.d

```diff
@@ -3075,6 +3075,13 @@ extern (C++) abstract class Type : RootObject
     {
         v.visit(this);
     }
+
+    final TypeFunction toTypeFunction()
+    {
+        if (ty != Tfunction)
+            assert(0);
```

Comment:
> assert(0) is lowered to a hlt/ud2/... in release mode, so it won't print any error message.

That's right. It's still there, though, doing its job, unlike `assert(ty == Tfunction)` which would just corrupt memory in release mode. (asserts should be left on in the compiler in release mode).

I put this assert in this way BECAUSE I was getting silent memory corruption, and who knows how long that has been going on. Fortunately, it would only happen if the compiler had already diagnosed errors, so at least it wasn't corrupting clean compiles.